lib/explorer/polars_backend/shared.ex

```diff
@@ -80,6 +80,21 @@ defmodule Explorer.PolarsBackend.Shared do
     end
   end
 
+  defp dtypes_match?(check_dtypes, out_dtypes) do
+    if map_size(check_dtypes) != map_size(out_dtypes) do
+      false
+    else
+      # The dtypes for each key must either be the same or `:unknown`.
+      Enum.reduce_while(check_dtypes, true, fn {key, check_value}, _ ->
```

Comment:
Maybe Enum.all? is enough here?

Reply:
Whoops yeah definitely [fc44836](https://github.com/elixir-explorer/explorer/pull/918/commits/fc44836909c9ebfee0e62b141cd427006c2f6acc)